dags/atd_executive_dashboard_expenses_revenue.py

```diff
@@ -0,0 +1,118 @@
+# test locally with: docker compose run --rm airflow-cli dags test atd_executive_dashboard_expenses_revenue
+
+import os
+
+from datetime import datetime, timedelta
+
+from airflow.decorators import task
+from airflow.models import DAG
+from airflow.operators.docker_operator import DockerOperator
+
+from utils.onepassword import get_env_vars_task
+from utils.slack_operator import task_fail_slack_alert
+
+DEPLOYMENT_ENVIRONMENT = os.getenv("ENVIRONMENT", "development")
+
+default_args = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2015, 12, 1),
+    "email_on_failure": False,
+    "email_on_retry": False,
+    "retries": 0,
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+docker_image = "atddocker/atd-executive-dashboard:production"
+
+REQUIRED_SECRETS = {
+    "SO_WEB": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.endpoint",
+    },
+    "SO_TOKEN": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.appToken",
+    },
+    "SO_SECRET": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeySecret",
+    },
+    "SO_KEY": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeyId",
+    },
+    "EXP_DATASET": {
+        "opitem": "Executive Dashboard",
+        "opfield": "datasets.Expenses",
+    },
+    "REV_DATASET": {
+        "opitem": "Executive Dashboard",
+        "opfield": "datasets.Revenue",
+    },
+    "BUCKET_NAME": {
+        "opitem": "Executive Dashboard",
+        "opfield": "s3.Bucket",
+    },
+    "AWS_ACCESS_KEY": {
+        "opitem": "Executive Dashboard",
+        "opfield": "s3.AWS Access Key",
+    },
+    "AWS_SECRET_ACCESS_KEY": {
+        "opitem": "Executive Dashboard",
+        "opfield": "s3.AWS Secret Access Key",
+    },
+    "BASE_URL": {
+        "opitem": "Microstrategy API",
+        "opfield": "shared.Base URL",
+    },
+    "PROJECT_ID": {
+        "opitem": "Microstrategy API",
+        "opfield": "shared.Project ID",
+    },
+    "MSTRO_USERNAME": {
+        "opitem": "Microstrategy API",
+        "opfield": "shared.Username",
+    },
+    "MSTRO_PASSWORD": {
+        "opitem": "Microstrategy API",
+        "opfield": "shared.Password",
+    },
+}
+
+with DAG(
+    dag_id="atd_executive_dashboard_expenses_revenue",
+    description="Downloads two Microstrategy Reports for Expenses and Revenue. \
+    Places the results as a CSV in a S3 bucket. \
+    Then publishes the data to a Socrata dataset",
+    default_args=default_args,
+    schedule_interval="00 11 * * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
+    dagrun_timeout=timedelta(minutes=120),
```

Comment:
Could this be too long? I didn't think so because it took about 15 min to run for me now, and its sounds like it is a full replace so it will grow over time. Just wanted to check in on it.